src/server/scripts/Commands/cs_anticheat.cpp

```diff
@@ -0,0 +1,264 @@
+            if (!player)
+            {
+                handler->PSendSysMessage(LANG_PLAYER_NOT_EXIST_OR_OFFLINE, strCommand.c_str());
+                handler->SetSentErrorMessage(true);
+                return false;
+            }
+            else
+                sAnticheatMgr->AnticheatDeleteCommand(player->GetGUID().GetCounter());
+        }
+
+        return true;
+    }
+
+    static bool HandleAntiCheatPlayerCommand(ChatHandler* handler, char const* args)
+    {
+        if (!sWorld->getBoolConfig(CONFIG_ANTICHEAT_ENABLE))
+        {
+            handler->SendSysMessage(LANG_COMMAND_ACDISABLED);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        char* command = strtok(const_cast<char*>(args), " ");
+
+        ObjectGuid::LowType lowGuid = 0;
+        Player* player = nullptr;
+
+        if (command)
+        {
+            std::string strCommand(command);
+
+            normalizePlayerName(strCommand);
+            player = ObjectAccessor::FindPlayerByName(strCommand); //get player by name
+
+            if (player)
+                lowGuid = player->GetGUID().GetCounter();
+        }
+        else
+        {
+            player = handler->getSelectedPlayer();
+            if (player)
+                lowGuid = player->GetGUID().GetCounter();
+        }
+
+        if (!lowGuid)
+        {
+            handler->SendSysMessage(LANG_PLAYER_NOT_FOUND);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        float average = sAnticheatMgr->GetAverage(lowGuid);
+        uint32 total_reports = sAnticheatMgr->GetTotalReports(lowGuid);
+        uint32 speed_reports = sAnticheatMgr->GetTypeReports(lowGuid, REPORT_TYPE_SPEED);
+        uint32 fly_reports = sAnticheatMgr->GetTypeReports(lowGuid, REPORT_TYPE_FLY);
+        uint32 jump_reports = sAnticheatMgr->GetTypeReports(lowGuid, REPORT_TYPE_JUMP);
+        uint32 waterwalk_reports = sAnticheatMgr->GetTypeReports(lowGuid, REPORT_TYPE_WATERWALK);
+        uint32 teleportplane_reports = sAnticheatMgr->GetTypeReports(lowGuid, REPORT_TYPE_TELEPORT_PLANE);
+        uint32 climb_reports = sAnticheatMgr->GetTypeReports(lowGuid, REPORT_TYPE_CLIMB);
+
+        handler->PSendSysMessage(LANG_COMMAND_ACPLAYER_INFO, player->GetName().c_str());
+        handler->PSendSysMessage(LANG_COMMAND_ACPLAYER_STATS, average, total_reports);
+        handler->PSendSysMessage(LANG_COMMAND_ACPLAYER_SPEEDFLYJUMP, speed_reports, fly_reports, jump_reports);
+        handler->PSendSysMessage(LANG_COMMAND_ACPLAYER_WATERPLANE, waterwalk_reports, teleportplane_reports);
+        handler->PSendSysMessage(LANG_COMMAND_ACPLAYER_CLIMB, climb_reports);
+        return true;
+    }
+
+    static bool HandleAntiCheatHandleCommand(ChatHandler* handler, char const* args)
+    {
+        if (!*args)
+            return false;
+
+        std::string strCommand = const_cast<char*>(args);
```

Comment:
doesnt look like a necessary cast. The value should be copied by the string constructor.